RecDP/pyrecdp/primitives/operations/text_deduplication.py

```diff
@@ -147,6 +150,95 @@ def process_spark(self, spark, spark_df: DataFrame) -> DataFrame:
+        super().__init__(settings)        
+        self.support_spark = True
+        self.support_ray = False
+        self.text_key = text_key
+        self.inplace = True
+        self.num_perm = num_perm
+        self.ngram_size = ngram_size
+        self.bands = bands
+        self.ranges = ranges
+    
+    def process_spark(self, spark, spark_df: DataFrame) -> DataFrame:
+        if self.inplace:
+            column_names = spark_df.columns
+            if 'norm_text' in column_names:
+                is_norm = False
+                spark_df = spark_df.select('norm_text', *[n for n in column_names if n != 'norm_text'])
+            else:
+                is_norm = True
+                spark_df = spark_df.select('text', *[n for n in column_names if n != 'text'])
+            
+            df_with_id = global_unique_id(spark_df, 'filename_docid')
+            pipeline = minHashLSH_prepare(df_with_id, self.num_perm, self.ngram_size, self.bands, self.ranges, is_norm)
+            with Timer("generate minHashLsh"):
+                results = pipeline.collect()
+                
+            with Timer("generate_connected_components => duplicates"):
+                components = generate_connected_components.generate_connected_components_py(results)
+                duplicates = [c for c_list in components for c in c_list[1:]]
+                R = Row('filename_docid')
+                total_dup = len(duplicates)
+                if total_dup != 0:
+                    duplicates_sdf = spark.createDataFrame([R(dup) for dup in duplicates]).cache()
+                    total_dup = duplicates_sdf.count()
+            if total_dup == 0:
+                from pyspark.sql.types import StructType, StructField, StringType
+                ret = spark.createDataFrame(spark.sparkContext.emptyRDD(), StructType([StructField('global_id', StringType(), False)]))
+            else:
+                with Timer("Get global_id for duplicated data"):
+                    ret = df_with_id.join(duplicates_sdf, 'filename_docid', 'inner').select('global_id').cache()
+            return ret
+            
+        else:
+            raise NotImplementedError("We only support inplace modification for FuzzyDeduplicate.")
+    
+LLMOPERATORS.register(FuzzyDeduplicateGenDict)
+
+class FuzzyDeduplicateApplyDict(BaseLLMOperation):
+    def __init__(self, text_key = 'text', num_perm = 256, ngram_size = 13, bands = 9, ranges = 13):
+        settings = {'text_key': text_key, 'num_perm': 256, 'ngram_size': 3, 'bands': 9, 'ranges': 13}
+        super().__init__(settings)        
+        self.support_spark = True
+        self.support_ray = False
+        self.text_key = text_key
+        self.inplace = True
+        self.num_perm = num_perm
+        self.ngram_size = ngram_size
+        self.bands = bands
+        self.ranges = ranges
+
+    def execute_spark(self, pipeline, rdp, child_ds=None, global_df=None):
```

Comment:
I don't think we need to overrise execute_spark since the super class function will pass *child_output to process_spark